native/core/src/models/status/membership.rs

```diff
@@ -1,147 +1,89 @@
-use matrix_sdk::ruma::{
-    events::room::member::MembershipChange as RumaMembershipChange, OwnedMxcUri, OwnedUserId,
+use matrix_sdk_base::ruma::{
+    events::room::member::MembershipChange as RumaMembershipChange, OwnedUserId,
 };
+use matrix_sdk_ui::timeline::{MembershipChange as SdkMembershipChange, RoomMembershipChange};
 use serde::{Deserialize, Serialize};
 
-/// A simple representation of a change, containing old and new data.
 #[derive(Clone, Debug, Serialize, Deserialize)]
-#[allow(clippy::exhaustive_structs)]
-pub struct Change<T> {
-    /// The old data.
-    pub old: T,
-
-    /// The new data.
-    pub new: T,
+pub struct MembershipContent {
+    user_id: OwnedUserId,
+    change: String,
 }
 
-/// Translation of the membership change in `m.room.member` event.
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub enum MembershipChangeType {
-    /// User joined the room.
-    Joined,
-
-    /// User left the room.
-    Left,
-
-    /// User was banned.
-    Banned,
-
-    /// User was unbanned.
-    Unbanned,
-
-    /// User was kicked.
-    Kicked,
-
-    /// User was invited.
-    Invited,
-
-    /// User was kicked and banned.
-    KickedAndBanned,
-
-    /// User accepted the invite.
-    InvitationAccepted,
-
-    /// User rejected the invite.
-    InvitationRejected,
-
-    /// User had their invite revoked.
-    InvitationRevoked,
-
-    /// User knocked.
-    Knocked,
-
-    /// User had their knock accepted.
-    KnockAccepted,
-
-    /// User retracted their knock.
-    KnockRetracted,
-
-    /// User had their knock denied.
-    KnockDenied,
-
-    /// `displayname` or `avatar_url` changed.
-    ProfileChanged {
-        /// The details of the displayname change, if applicable.
-        displayname_change: Option<Change<Option<String>>>,
+impl TryFrom<&RoomMembershipChange> for MembershipContent {
+    type Error = ();
 
-        /// The details of the avatar url change, if applicable.
-        avatar_url_change: Option<Change<Option<OwnedMxcUri>>>,
-    },
+    fn try_from(value: &RoomMembershipChange) -> Result<Self, Self::Error> {
```

Comment:
I thought I mentioned before that I'd rather we keep the enum type, either a local or the SDK version, and only return the string on the API level - internally we'd like to keep the actual type and ideally in the future uniffi API we can also return that. internally we don't want to use strings. convert-late is the goal.

Reply:
As you can see [here](https://github.com/matrix-org/matrix-rust-sdk/blob/c8536e9e46cc99b946db50f20838a0de928ee143/crates/matrix-sdk-ui/src/timeline/event_item/content/mod.rs#L587-L592), `RoomMembershipChange` doesn't have `Serialize` and `Deserialize` derives.
Without them, we can't save this type to cache store.

Also I hoped to reuse the types from `matrix-rust-sdk` as much as possible.
For example, I am going to reuse `RoomNameEventContent` from `matrix-rust-sdk` in future PR for room state implementation.
But `RoomMembershipChange` doesn't have `Serialize` and `Deserialize` derives now so we can't reuse it.
We have to define another struct.